Userland/Applications/NetworkSettings/NetworkSettingsWidget.cpp

```diff
@@ -134,27 +138,64 @@ void NetworkSettingsWidget::on_switch_enabled_or_dhcp()
 
 void NetworkSettingsWidget::apply_settings()
 {
-    auto config_file = Core::ConfigFile::open_for_system("Network", Core::ConfigFile::AllowWriting::Yes).release_value_but_fixme_should_propagate_errors();
+    auto result = pass_changes_to_helper();
+    if (result.is_error()) {
+        GUI::MessageBox::show_error(window(), result.release_error().string_literal());
+        return;
+    }
+}
+
+ErrorOr<void> NetworkSettingsWidget::pass_changes_to_helper()
```

Comment:
I am not a particular fan of this, at least the name. At one point we will convert the virtual `apply_settings()` to return `ErrorOr<T>`, and then this would need a refactor. If you want to keep it at least call it `apply_settings_impl()` which is the pattern we use elsewhere.